src/file-utils/edfi_lms_file_utils/file_reader.py

```diff
@@ -76,6 +76,44 @@ def _read_csv(
     return _default()
 
 
+def read_file_from_path(
+    full_path: str, resource_name: Optional[str] = None, nrows: Optional[int] = None
+) -> pd.DataFrame:
+    """
+    Reads the CSV file for the given path into a Pandas DataFrame.
+
+    Parameters
+    ----------
+    full_path: str
+        The full path of the file.
+    resource_name: str
+        (Optional) The name of the resource for parsing DataFrame columns.
+    nrows: int or None
+        (Optional) number of rows to read from the file - useful for testing
+        without reading the entirety of a large file.
+
+    Returns
+    -------
+    Pandas DataFrame with columns matching the model definition / CSV file.
+
+    Notes
+    -----
+    All the valid resource_names are available in edfi_lms_file_utils.constants.Resources
+    """
+
+# If it is required to add extra_date_columns, those can be added conditionally based on the
+# resource_name
+    data_types = dict()
+    if resource_name == Resources.USERS:
+        data_types = DataTypes.USERS
+
+    if resource_name == Resources.SECTIONS:
+        data_types = DataTypes.SECTIONS
+
+    _read_csv(full_path, nrows, data_types)
+    return _default()
```

Comment:
Let's make sure this function is well covered by unit tests, particularly since it has branching conditions.